app/main.py

```diff
@@ -1 +1,28 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(
+            self, name: str, health: int = 100, hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, herbivore: Herbivore) -> None:
+        if isinstance(herbivore, Herbivore) and not herbivore.hidden:
+            herbivore.health -= 50
+            if herbivore.health <= 0:
+                Animal.alive.remove(herbivore)
```

Comment:
The logic of taking damage seems more general than only for Herbivore. It would be better to move it to the Animal class and create an Animal method like "take_damage".